packages/block-library/src/cover/edit.native.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Icon,
+	ImageWithFocalPoint,
+	PanelBody,
+	RangeControl,
+} from '@wordpress/components';
+import {
+	InnerBlocks,
+	InspectorControls,
+	MEDIA_TYPE_IMAGE,
+	MediaPlaceholder,
+	withColors,
+} from '@wordpress/block-editor';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { cover as icon } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import { attributesFromMedia } from './utils';
+import {
+	COVER_MIN_HEIGHT,
+	IMAGE_BACKGROUND_TYPE,
+	VIDEO_BACKGROUND_TYPE,
+} from './shared';
+
+/**
+ * Constants
+ */
+const ALLOWED_MEDIA_TYPES = [ MEDIA_TYPE_IMAGE ];
+const INNER_BLOCKS_TEMPLATE = [
+	[
+		'core/paragraph',
+		{
+			align: 'center',
+			placeholder: __( 'Write title…' ),
+		},
+	],
+];
+const COVER_MAX_HEIGHT = 1000;
+const COVER_DEFAULT_HEIGHT = 300;
+
+const Cover = ( {
+	attributes,
+	isAncestorSelected,
+	isParentSelected,
+	isSelected,
+	onFocus,
+	overlayColor,
+	setAttributes,
+} ) => {
+	const {
+		backgroundType,
+		dimRatio,
+		focalPoint,
+		gradientValue,
+		minHeight,
+		url,
+	} = attributes;
+	const CONTAINER_HEIGHT = minHeight || COVER_DEFAULT_HEIGHT;
+
+	// Used to set a default color for its InnerBlocks
+	// since there's no system to inherit styles yet
+	// the RichText component will check if there are
+	// parent styles for the current block. If there are,
+	// it will use that color instead.
+	useEffect( () => {
+		setAttributes( { childrenStyles: styles.defaultColor } );
+	}, [ setAttributes ] );
+
+	const onSelectMedia = ( media ) => {
+		const onSelect = attributesFromMedia( setAttributes );
+		onSelect( media );
+	};
+
+	const onHeightChange = ( value ) => {
+		if ( minHeight || value !== COVER_DEFAULT_HEIGHT ) {
+			setAttributes( { minHeight: value } );
+		}
+	};
+
+	const onOpactiyChange = ( value ) => {
+		setAttributes( { dimRatio: value } );
+	};
+
+	const getOpacity = () => {
+		// Set opacity to 1 while video support is not available
+		if ( VIDEO_BACKGROUND_TYPE === backgroundType ) {
+			return 1;
+		}
+
+		return url ? dimRatio / 100 : 1;
+	};
+
+	const getOverlayStyles = () => {
```

Comment:
For this and `getOverlay`, what's the benefit of defining a function versus just assigning a `opacity`/`overlayStyles` const?

Reply:
After a few changes, I updated it to be in a const within `getOverlayStyles`

Reply:
I'm still not sure why it is a function, since it's only called once from within the same component.

Reply:
You're right! Didn't see it in your first comment sorry about that! Updated https://github.com/WordPress/gutenberg/pull/19722/commits/ad28977d821e3b66350523433e4edb57f1dab991#diff-b17d53802519ed746057fb2ee4a915edR113